image_tools/src/cam2image.cpp

```diff
@@ -105,26 +105,26 @@ int main(int argc, char * argv[])
 
   // Set the parameters of the quality of service profile. Initialize as the default profile
   // and set the QoS parameters specified on the command line.
-  rmw_qos_profile_t custom_camera_qos_profile = rmw_qos_profile_default;
+  rclcpp::PublisherOptions<> custom_camera_qos_profile;
 
   // Depth represents how many messages to store in history when the history policy is KEEP_LAST.
-  custom_camera_qos_profile.depth = depth;
+  custom_camera_qos_profile.qos_history_depth(depth);
 
   // The reliability policy can be reliable, meaning that the underlying transport layer will try
   // ensure that every message gets received in order, or best effort, meaning that the transport
   // makes no guarantees about the order or reliability of delivery.
-  custom_camera_qos_profile.reliability = reliability_policy;
+  custom_camera_qos_profile.qos_profile().reliability = reliability_policy;
 
   // The history policy determines how messages are saved until the message is taken by the reader.
   // KEEP_ALL saves all messages until they are taken.
   // KEEP_LAST enforces a limit on the number of messages that are saved, specified by the "depth"
   // parameter.
-  custom_camera_qos_profile.history = history_policy;
+  custom_camera_qos_profile.qos_profile().history = history_policy;
 
   RCLCPP_INFO(node_logger, "Publishing data on topic '%s'", topic.c_str());
   // Create the image publisher with our custom QoS profile.
-  auto pub = node->create_publisher<sensor_msgs::msg::Image>(
-    topic, custom_camera_qos_profile);
+  auto pub = node->create_publisher<sensor_msgs::msg::Image>(topic, nullptr,
```

Comment:
If something is getting passed in as a `nullptr` should it also be moved to the Options class?